packages/mui-base/src/useAutocomplete/useAutocomplete.js

```diff
@@ -236,7 +236,15 @@ export default function useAutocomplete(props) {
   });
 
   React.useEffect(() => {
-    const valueChange = value !== previousProps.value;
+    // The value can be an instance of object and thus comparing it with basic equality lead to
+    // invalid results. Instead we should compare its final label to detect a value change.
+    const valueChange =
+      value !== previousProps.value &&
+      (previousProps.value === undefined ||
+        previousProps.value === null ||
+        value === undefined ||
+        value === null ||
```

Comment:
This case is required because of `handleValue(event, multiple ? [] : null, 'clear');`. It resets `value` (possibly undefined or string) to null.